src/cart/src/cart/crt_ctl.c

```diff
@@ -81,31 +82,27 @@ crt_ctl_fill_buffer_cb(d_list_t *rlink, void *arg)
 	struct crt_uri_item	*ui;
 	int			*idx;
 	struct crt_uri_cache	*uri_cache = arg;
+	crt_phy_addr_t		 uri;
 	int			 i;
 	int			 rc = 0;
 
 	D_ASSERT(rlink != NULL);
 	D_ASSERT(arg != NULL);
 
 	ui = crt_ui_link2ptr(rlink);
-
-	D_MUTEX_LOCK(&ui->ui_mutex);
-
 	idx = &uri_cache->idx;
-
 	for (i = 0; i < CRT_SRV_CONTEXT_NUM; i++) {
-		if (ui->ui_uri[i] == NULL)
+		uri = atomic_load_consume(&ui->ui_uri[i]);
```

Comment:
This maps to atomic_load_explicit(ptr, memory_order_consume) but I think for reference counting and these URLs we want atomic_load_explicit(ptr, memory_order_relaxed) because we don't care about memory ordering here and don't want to stall the memory pipeline.

Reply:
ok